### PR TITLE
fix(gainsignt_px): undefined attributes issue

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change
https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/642245a1d802040008336b17?event_id=642245a100ba5d787ceb0000&i=wh&m=nw

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/gainsight_px/transform.js [hunk 1]
```diff
@@ -235,13 +235,13 @@ const trackResponseBuilder = (message, { Config }) => {
...
 
-  if (payload.attributes && payload.attributes.globalContext) {
+  if (payload?.attributes && payload?.attributes?.globalContext) {
     delete payload.attributes.globalContext;
   }
 
   payload = {
     ...payload,
-    attributes: formatEventProps(payload.attributes),
+    attributes: payload?.attributes ? formatEventProps(payload.attributes) : {},
```
Comment (ItsSudip) [inline src/v0/destinations/gainsight_px/transform.js:244]:
```suggestion
    attributes: payload.attributes ? formatEventProps(payload.attributes) : {},
```

File: src/v0/destinations/gainsight_px/transform.js [hunk 2]
```diff
@@ -235,13 +235,13 @@ const trackResponseBuilder = (message, { Config }) => {
     globalContext = getHashFromArray(Config.globalContextMap, 'from', 'to', false);
   }
 
-  if (payload.attributes && payload.attributes.globalContext) {
+  if (payload?.attributes && payload?.attributes?.globalContext) {
```
Comment (ItsSudip) [inline src/v0/destinations/gainsight_px/transform.js:238]:
```suggestion
  if (payload.attributes?.globalContext) {
```

Comment (mihir-4116) [reply src/v0/destinations/gainsight_px/transform.js]: updated

File: src/v0/destinations/gainsight_px/transform.js (hunk 1 above)
Comment (mihir-4116) [reply src/v0/destinations/gainsight_px/transform.js]: updated

**Review by ItsSudip — APPROVED** [review]

**Review by anantjain45823 — APPROVED** [review]

**Pull request merged**